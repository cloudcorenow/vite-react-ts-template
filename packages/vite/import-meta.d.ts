import type {
  ImportMeta as ViteImportMeta,
  ImportMetaEnv as ViteImportMetaEnv
} from './client/types'

declare global {
  interface ImportMeta extends ViteImportMeta {}
  interface ImportMetaEnv extends ViteImportMetaEnv {}
}
