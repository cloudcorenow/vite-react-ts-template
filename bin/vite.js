#!/usr/bin/env node
const argv = require('minimist')(process.argv.slice(2))
const server = require('../dist/server').createServer(argv)
const getIPv4AddressList = require('../dist/server/utils').getIPv4AddressList

let port = argv.port || 3000

server.on('error', (e) => {
  if (e.code === 'EADDRINUSE') {
    console.log(`port ${port} is in use, trying another one...`)
    setTimeout(() => {
      server.close()
      server.listen(++port)
    }, 100)
  } else {
    console.error(e)
  }
})

server.on('listening', () => {
  console.log(`Running at:`)
  getIPv4AddressList().forEach((ip) => {
    console.log(`  > http://${ip}:${port}`)
  })
  console.log(' ')
})

server.listen(port)
