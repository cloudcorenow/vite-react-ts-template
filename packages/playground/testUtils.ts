// test utils used in e2e tests for playgrounds.
// this can be directly imported in any playground tests as 'testUtils', e.g.
// `import { getColor } from 'testUtils'`

import fs from 'fs-extra'
import path from 'path'
import colors from 'css-color-names'
import { ElementHandle } from 'playwright-chromium'
import expectExport from 'expect'
import * as http from 'http'
import sirv from 'sirv'
import {
  createServer,
  build,
  ViteDevServer,
  UserConfig,
  PluginOption,
  ResolvedConfig,
  Logger,
  Manifest
} from 'vite'
import { Page, Browser, chromium } from 'playwright-chromium'
// eslint-disable-next-line node/no-extraneous-import
import { RollupError, RollupWatcher, RollupWatcherEvent } from 'rollup'
import { suite, Test } from 'uvu'

export function slash(p: string): string {
  return p.replace(/\\/g, '/')
}

export let testDir: string
export const isBuild = !!process.env.VITE_TEST_BUILD
export const workspaceRoot = path.resolve(__dirname, '../../')

// injected by the test env
declare global {
  const page: Page | undefined
  const browser: Browser
  const expect: typeof expectExport

  const browserLogs: string[]
  const serverLogs: string[]
  const viteTestUrl: string | undefined
  const watcher: RollupWatcher | undefined
  let beforeAllError: Error | null // error caught in beforeAll, useful if you want to test error scenarios on build
}

declare const global: {
  page?: Page
  browser: Browser
  expect: typeof expectExport

  browserLogs: string[]
  serverLogs: string[]
  viteTestUrl?: string
  watcher?: RollupWatcher
  beforeAllError: Error | null
}

let server: ViteDevServer | http.Server
let tempDir: string
let rootDir: string

const setBeforeAllError = (err) => ((global as any).beforeAllError = err)
const getBeforeAllError = () => (global as any).beforeAllError
//init with null so old errors don't carry over
setBeforeAllError(null)

global.browserLogs = []
const onConsole = (msg) => {
  global.browserLogs.push(msg.text())
}

global.expect = expectExport

export async function uvuSetup(testPath: string) {
  global.browser = await chromium.launch({ headless: true })
  const page = (global.page = await global.browser.newPage())

  const testName = slash(testPath).match(/playground\/([\w-]+)\//)[1]
  testDir = path.resolve(__dirname, '../temp', testName)
  global.browserLogs = []

  try {
    page.on('console', onConsole)

    // if this is a test placed under playground/xxx/__tests__
    // start a vite server in that directory.
    if (testName) {
      tempDir = path.resolve(__dirname, '../temp/', testName)

      // when `root` dir is present, use it as vite's root
      const testCustomRoot = path.resolve(tempDir, 'root')
      rootDir = fs.existsSync(testCustomRoot) ? testCustomRoot : tempDir

      const testCustomServe = path.resolve(testPath, 'serve.cjs')
      if (fs.existsSync(testCustomServe)) {
        // test has custom server configuration.
        const { serve, preServe } = require(testCustomServe)
        if (preServe) {
          await preServe(rootDir, isBuild)
        }
        if (serve) {
          server = await serve(rootDir, isBuild)
          return
        }
      }

      const serverLogs: string[] = []

      const options: UserConfig = {
        root: rootDir,
        logLevel: 'silent',
        server: {
          watch: {
            // During tests we edit the files too fast and sometimes chokidar
            // misses change events, so enforce polling for consistency
            usePolling: true,
            interval: 100
          },
          host: true,
          fs: {
            strict: !isBuild
          }
        },
        build: {
          // skip transpilation during tests to make it faster
          target: 'esnext'
        },
        customLogger: createInMemoryLogger(serverLogs)
      }

      global.serverLogs = serverLogs

      if (!isBuild) {
        process.env.VITE_INLINE = 'inline-serve'
        server = await (await createServer(options)).listen()
        // use resolved port/base from server
        const base = server.config.base === '/' ? '' : server.config.base
        const url =
          (global.viteTestUrl = `http://localhost:${server.config.server.port}${base}`)
        await page.goto(url)
      } else {
        process.env.VITE_INLINE = 'inline-build'
        // determine build watch
        let resolvedConfig: ResolvedConfig
        const resolvedPlugin: () => PluginOption = () => ({
          name: 'vite-plugin-watcher',
          configResolved(config) {
            resolvedConfig = config
          }
        })
        options.plugins = [resolvedPlugin()]
        const rollupOutput = await build(options)
        const isWatch = !!resolvedConfig!.build.watch
        // in build watch,call startStaticServer after the build is complete
        if (isWatch) {
          global.watcher = rollupOutput as RollupWatcher
          await notifyRebuildComplete(global.watcher)
        }
        const url = (global.viteTestUrl = await startStaticServer())
        await page.goto(url)
      }
    }
  } catch (e) {
    // jest doesn't exit if our setup has error here
    // https://github.com/facebook/jest/issues/2713
    setBeforeAllError(e)

    // Closing the page since an error in the setup, for example a runtime error
    // when building the playground should skip further tests.
    // If the page remains open, a command like `await page.click(...)` produces
    // a timeout with an exception that hides the real error in the console.
    await page.close()
  }
}

export async function uvuReset() {
  global.browserLogs = []
  global.page?.off('console', onConsole)
  global.serverLogs = []
  global.watcher?.close()
  await global.page?.close()
  await server?.close()
  await global.browser?.close()
  const beforeAllErr = getBeforeAllError()
  if (beforeAllErr) {
    throw beforeAllErr
  }
}

export function describe(name: string, fn: (test: Test) => void) {
  const s = suite(name)
  fn(s)
  s.run()
}

function startStaticServer(): Promise<string> {
  // check if the test project has base config
  const configFile = path.resolve(rootDir, 'vite.config.js')
  let config: UserConfig
  try {
    config = require(configFile)
  } catch (e) {}
  const base = (config?.base || '/') === '/' ? '' : config.base

  // @ts-ignore
  if (config && config.__test__) {
    // @ts-ignore
    config.__test__()
  }

  // start static file server
  const serve = sirv(path.resolve(rootDir, 'dist'))
  const httpServer = (server = http.createServer((req, res) => {
    if (req.url === '/ping') {
      res.statusCode = 200
      res.end('pong')
    } else {
      serve(req, res)
    }
  }))
  let port = 5000
  return new Promise((resolve, reject) => {
    const onError = (e: any) => {
      if (e.code === 'EADDRINUSE') {
        httpServer.close()
        httpServer.listen(++port)
      } else {
        reject(e)
      }
    }
    httpServer.on('error', onError)
    httpServer.listen(port, () => {
      httpServer.removeListener('error', onError)
      resolve(`http://localhost:${port}${base}`)
    })
  })
}

/**
 * Send the rebuild complete message in build watch
 */
export async function notifyRebuildComplete(
  watcher: RollupWatcher
): Promise<RollupWatcher> {
  let callback: (event: RollupWatcherEvent) => void
  await new Promise((resolve, reject) => {
    callback = (event) => {
      if (event.code === 'END') {
        resolve(true)
      }
    }
    watcher.on('event', callback)
  })
  return watcher.removeListener('event', callback)
}

function createInMemoryLogger(logs: string[]): Logger {
  const loggedErrors = new WeakSet<Error | RollupError>()
  const warnedMessages = new Set<string>()

  const logger: Logger = {
    hasWarned: false,
    hasErrorLogged: (err) => loggedErrors.has(err),
    clearScreen: () => {},
    info(msg) {
      logs.push(msg)
    },
    warn(msg) {
      logs.push(msg)
      logger.hasWarned = true
    },
    warnOnce(msg) {
      if (warnedMessages.has(msg)) return
      logs.push(msg)
      logger.hasWarned = true
      warnedMessages.add(msg)
    },
    error(msg, opts) {
      logs.push(msg)
      if (opts?.error) {
        loggedErrors.add(opts.error)
      }
    }
  }

  return logger
}

const hexToNameMap: Record<string, string> = {}
Object.keys(colors).forEach((color) => {
  hexToNameMap[colors[color]] = color
})

function componentToHex(c: number): string {
  const hex = c.toString(16)
  return hex.length === 1 ? '0' + hex : hex
}

function rgbToHex(rgb: string): string {
  const match = rgb.match(/rgb\((\d+),\s*(\d+),\s*(\d+)\)/)
  if (match) {
    const [_, rs, gs, bs] = match
    return (
      '#' +
      componentToHex(parseInt(rs, 10)) +
      componentToHex(parseInt(gs, 10)) +
      componentToHex(parseInt(bs, 10))
    )
  } else {
    return '#000000'
  }
}

const timeout = (n: number) => new Promise((r) => setTimeout(r, n))

async function toEl(el: string | ElementHandle): Promise<ElementHandle> {
  if (typeof el === 'string') {
    return await page.$(el)
  }
  return el
}

export async function getColor(el: string | ElementHandle): Promise<string> {
  el = await toEl(el)
  const rgb = await el.evaluate((el) => getComputedStyle(el as Element).color)
  return hexToNameMap[rgbToHex(rgb)] || rgb
}

export async function getBg(el: string | ElementHandle): Promise<string> {
  el = await toEl(el)
  return el.evaluate((el) => getComputedStyle(el as Element).backgroundImage)
}

export function readFile(filename: string): string {
  return fs.readFileSync(path.resolve(testDir, filename), 'utf-8')
}

export function editFile(
  filename: string,
  replacer: (str: string) => string,
  runInBuild: boolean = false
): void {
  if (isBuild && !runInBuild) return
  filename = path.resolve(testDir, filename)
  const content = fs.readFileSync(filename, 'utf-8')
  const modified = replacer(content)
  fs.writeFileSync(filename, modified)
}

export function addFile(filename: string, content: string): void {
  fs.writeFileSync(path.resolve(testDir, filename), content)
}

export function removeFile(filename: string): void {
  fs.unlinkSync(path.resolve(testDir, filename))
}

export function listAssets(base = ''): string[] {
  const assetsDir = path.join(testDir, 'dist', base, 'assets')
  return fs.readdirSync(assetsDir)
}

export function findAssetFile(match: string | RegExp, base = ''): string {
  const assetsDir = path.join(testDir, 'dist', base, 'assets')
  const files = fs.readdirSync(assetsDir)
  const file = files.find((file) => {
    return file.match(match)
  })
  return file ? fs.readFileSync(path.resolve(assetsDir, file), 'utf-8') : ''
}

export function readManifest(base = ''): Manifest {
  return JSON.parse(
    fs.readFileSync(path.join(testDir, 'dist', base, 'manifest.json'), 'utf-8')
  )
}

/**
 * Poll a getter until the value it returns includes the expected value.
 */
export async function untilUpdated(
  poll: () => string | Promise<string>,
  expected: string,
  runInBuild = false
): Promise<void> {
  if (isBuild && !runInBuild) return
  const maxTries = process.env.CI ? 100 : 50
  for (let tries = 0; tries < maxTries; tries++) {
    const actual = (await poll()) || ''
    if (actual.indexOf(expected) > -1 || tries === maxTries - 1) {
      expect(actual).toMatch(expected)
      break
    } else {
      await timeout(50)
    }
  }
}
