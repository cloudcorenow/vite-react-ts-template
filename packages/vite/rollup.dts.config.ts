import { readFileSync } from 'node:fs'
import { fileURLToPath } from 'node:url'
import { type Plugin, defineConfig } from 'rollup'
import dts from 'rollup-plugin-dts'
import { parse } from '@babel/parser'
import type { Node } from '@babel/types'
import { walk } from 'estree-walker'
import MagicString from 'magic-string'

const depTypesDir = new URL('./src/types/', import.meta.url)
const pkg = JSON.parse(
  readFileSync(new URL('./package.json', import.meta.url)).toString(),
)

export default defineConfig({
  input: './temp/node/index.d.ts',
  output: {
    file: './dist/node/index.d.ts',
    format: 'es',
  },
  external: [
    /^node:*/,
    ...Object.keys(pkg.dependencies),
    // lightningcss types are bundled
    ...Object.keys(pkg.devDependencies).filter((d) => d !== 'lightningcss'),
  ],
  plugins: [patchTypes(), dts({ respectExternal: true })],
  onwarn(warning, warn) {
    // `@internal` props could refer to imported values, but since they are stripped
    // the imported values will be seemingly unused. Ignore these warnings.
    if (warning.code === 'UNUSED_EXTERNAL_IMPORT') return
    warn(warning)
  },
})

/**
 * Patch the types file before passing to dts plugin
 * 1. Resolve `dep-types/*` and `types/*` imports
 * 2. Remove `@internal` declarations
 * 3. Remove unnecessary comments
 * 4. Validate unallowed dependency imports
 */
function patchTypes(): Plugin {
  return {
    name: 'patch-types',
    resolveId(id) {
      // Dep types should be bundled
      if (id.startsWith('dep-types/')) {
        const fileUrl = new URL(
          `./${id.slice('dep-types/'.length)}.d.ts`,
          depTypesDir,
        )
        return fileURLToPath(fileUrl)
      }
      // Ambient types are unbundled and externalized
      if (id.startsWith('types/')) {
        return {
          id: '../../' + (id.endsWith('.js') ? id : id + '.js'),
          external: true,
        }
      }
    },
    transform(code, id) {
      if (id.includes('/node_modules/')) return

      const s = new MagicString(code)
      const ast = parse(code, {
        plugins: ['typescript'],
        sourceType: 'module',
      })

      if (code.includes('@internal')) {
        walk(ast as any, {
          enter(node: any) {
            if (removeInternal(s, node)) {
              this.skip()
            }
          },
        })
      }

      if (ast.comments?.length) {
        for (const comment of ast.comments) {
          if (
            // Remove unnecessary single-line comments
            comment.type === 'CommentLine' ||
            // Remove custom license code
            comment.value.includes('The MIT License')
          ) {
            // @ts-expect-error exists
            s.remove(comment.start, comment.end)
          }
        }
      }

      code = s.toString()

      if (code.includes('@internal')) {
        throw new Error(`Unhandled @internal declarations detected in ${id}`)
      }

      return code
    },
    renderChunk(_, chunk) {
      const deps = new Set(Object.keys(pkg.dependencies))
      // Validate that chunk imports do not import dev deps
      for (const id of chunk.imports) {
        if (
          !id.startsWith('./') &&
          !id.startsWith('../') &&
          !id.startsWith('node:') &&
          !deps.has(id)
        ) {
          // If validation failed, only warn and set exit code 1 so that files
          // are written to disk for inspection, but the build will fail
          this.warn(`${chunk.fileName} imports "${id}" which is not allowed`)
          process.exitCode = 1
        }
      }
    },
  }
}

// Reference: https://github.com/vuejs/core/blob/main/rollup.dts.config.js
function removeInternal(s: MagicString, node: Node): boolean {
  if (
    node.leadingComments &&
    node.leadingComments.some((c) => {
      return c.type === 'CommentBlock' && c.value.includes('@internal')
    })
  ) {
    const n = node as any
    let id: string | undefined
    if (n.id && n.id.type === 'Identifier') {
      id = n.id.name
    } else if (n.key && n.key.type === 'Identifier') {
      id = n.key.name
    }
    if (id) {
      s.overwrite(
        // @ts-expect-error exists
        node.leadingComments[0].start,
        node.end,
        `/* removed internal: ${id} */`,
      )
    } else {
      // @ts-expect-error exists
      s.remove(node.leadingComments[0].start, node.end)
    }
    return true
  }
  return false
}
