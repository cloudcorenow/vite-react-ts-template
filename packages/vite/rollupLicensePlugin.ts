import fs from 'node:fs'
import license from 'rollup-plugin-license'
import type { Dependency } from 'rollup-plugin-license'
import colors from 'picocolors'
import type { Plugin } from 'rollup'

export default function licensePlugin(
  licenseFilePath: string,
  licenseTitle: string,
  packageName: string,
): Plugin {
  return license({
    thirdParty(dependencies) {
      // https://github.com/rollup/rollup/blob/master/build-plugins/generate-license-file.js
      // MIT Licensed https://github.com/rollup/rollup/blob/master/LICENSE-CORE.md
      const coreLicense = fs.readFileSync(
        new URL('../../LICENSE', import.meta.url),
      )

      const deps = sortDependencies(dependencies)
      const licenses = sortLicenses(
        new Set(
          dependencies.map((dep) => dep.license).filter(Boolean) as string[],
        ),
      )

      let dependencyLicenseTexts = ''
      for (let i = 0; i < deps.length; i++) {
        // Find dependencies with the same license text so it can be shared
        const licenseText = deps[i].licenseText
        const sameDeps = [deps[i]]
        if (licenseText) {
          for (let j = i + 1; j < deps.length; j++) {
            if (licenseText === deps[j].licenseText) {
              sameDeps.push(...deps.splice(j, 1))
              j--
            }
          }
        }

        let text = `## ${sameDeps.map((d) => d.name).join(', ')}\n`

        for (let j = 0; j < sameDeps.length; j++) {
          const { license, author, maintainers, contributors, repository } =
            sameDeps[j]

          if (license) {
            text += `License: ${license}\n`
          }
          const names = new Set<string>()
          for (const person of [author, ...maintainers, ...contributors]) {
            const name = typeof person === 'string' ? person : person?.name
            if (name) {
              names.add(name)
            }
          }
          if (names.size > 0) {
            text += `By: ${Array.from(names).join(', ')}\n`
          }
          if (repository) {
            text += `Repository: ${
              typeof repository === 'string' ? repository : repository.url
            }\n`
          }
          if (j !== sameDeps.length - 1) {
            text += '\n'
          }
        }

        if (licenseText) {
          text +=
            '\n' +
            licenseText
              .trim()
              .replace(/\r\n|\r/g, '\n')
              .split('\n')
              .map((line) => `> ${line}`)
              .join('\n') +
            '\n'
        }

        if (i !== deps.length - 1) {
          text += '\n---------------------------------------\n\n'
        }

        dependencyLicenseTexts += text
      }

      const licenseText =
        `# ${licenseTitle}\n` +
        `${packageName} is released under the MIT license:\n\n` +
        coreLicense +
        `\n# Licenses of bundled dependencies\n` +
        `The published ${packageName} artifact additionally contains code with the following licenses:\n` +
        `${licenses.join(', ')}\n\n` +
        `# Bundled dependencies:\n` +
        dependencyLicenseTexts

      const existingLicenseText = fs.readFileSync(licenseFilePath, 'utf-8')
      if (existingLicenseText !== licenseText) {
        fs.writeFileSync(licenseFilePath, licenseText)
        console.warn(
          colors.yellow(
            '\nLICENSE.md updated. You should commit the updated file.\n',
          ),
        )
      }
    },
  })
}

function sortDependencies(dependencies: Dependency[]) {
  return dependencies.sort(({ name: nameA }, { name: nameB }) => {
    return nameA! > nameB! ? 1 : nameB! > nameA! ? -1 : 0
  })
}

function sortLicenses(licenses: Set<string>) {
  let withParenthesis: string[] = []
  let noParenthesis: string[] = []
  licenses.forEach((license) => {
    if (/^\(/.test(license)) {
      withParenthesis.push(license)
    } else {
      noParenthesis.push(license)
    }
  })
  withParenthesis = withParenthesis.sort()
  noParenthesis = noParenthesis.sort()
  return [...noParenthesis, ...withParenthesis]
}
