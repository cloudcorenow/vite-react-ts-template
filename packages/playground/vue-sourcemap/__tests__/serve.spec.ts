import { fromComment } from 'convert-source-map'
import { normalizePath } from 'vite'
import { isBuild, testDir } from 'testUtils'

if (!isBuild) {
  const root = normalizePath(testDir)

  const getStyleTagContentIncluding = async (content: string) => {
    const styles = await page.$$('style')
    for (const style of styles) {
      const text = await style.textContent()
      if (text.includes(content)) {
        return text
      }
    }
    throw new Error('Not found')
  }

  const extractSourcemap = (content: string) =>
    fromComment(content.trim().split('\n').at(-1)).toObject()

  const assertSourcemap = (map: any, { sources }: { sources: string[] }) => {
    expect(map.sources).toStrictEqual(sources)
    expect(map.mappings).toMatchSnapshot('mappings')
    expect(map.sourcesContent).toMatchSnapshot('sourcesContent')
  }

  test('css', async () => {
    const css = await getStyleTagContentIncluding('.css ')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [`${root}/Css.vue`]
    })
  })

  test('css module', async () => {
    const css = await getStyleTagContentIncluding('._css-module_')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [`${root}/Css.vue`]
    })
  })

  test('css scoped', async () => {
    const css = await getStyleTagContentIncluding('.css-scoped[data-v-')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [`${root}/Css.vue`]
    })
  })

  test('sass', async () => {
    const css = await getStyleTagContentIncluding('.sass ')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [`${root}/Sass.vue`]
    })
  })

  test('sass with import', async () => {
    const css = await getStyleTagContentIncluding('.sass-with-import ')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [
        `${root}/sassWithImportImported.sass`,
        `${root}/SassWithImport.vue`
      ]
    })
  })

  test('less with additionalData', async () => {
    const css = await getStyleTagContentIncluding('.less ')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [`${root}/Less.vue`]
    })
  })

  test('src imported', async () => {
    const css = await getStyleTagContentIncluding('.src-import[data-v-')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [`${root}/src-import/src-import.css`]
    })
  })

  test('src imported sass', async () => {
    const css = await getStyleTagContentIncluding('.src-import-sass[data-v-')
    const map = extractSourcemap(css)
    assertSourcemap(map, {
      sources: [
        `${root}/src-import/src-import-imported.sass`,
        `${root}/src-import/src-import.sass`
      ]
    })
  })
}
