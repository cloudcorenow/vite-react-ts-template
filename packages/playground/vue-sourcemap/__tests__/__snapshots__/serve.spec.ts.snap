// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css module: mappings 1`] = `";AAaA;EACE,UAAU;AACZ"`;

exports[`css module: sourcesContent 1`] = `
Array [
  "<template>
  <p class=\\"css\\">&lt;css&gt;</p>
  <p :class=\\"$style['css-module']\\">&lt;css&gt; module</p>
  <p class=\\"css-scoped\\">&lt;css&gt; scoped</p>
</template>

<style>
.css {
  color: red;
}
</style>

<style module>
.css-module {
  color: red;
}
</style>

<style scoped>
.css-scoped {
  color: red;
}
</style>
",
]
`;

exports[`css scoped: mappings 1`] = `";AAmBA;EACE,UAAU;AACZ"`;

exports[`css scoped: sourcesContent 1`] = `
Array [
  "<template>
  <p class=\\"css\\">&lt;css&gt;</p>
  <p :class=\\"$style['css-module']\\">&lt;css&gt; module</p>
  <p class=\\"css-scoped\\">&lt;css&gt; scoped</p>
</template>

<style>
.css {
  color: red;
}
</style>

<style module>
.css-module {
  color: red;
}
</style>

<style scoped>
.css-scoped {
  color: red;
}
</style>
",
]
`;

exports[`css: mappings 1`] = `";AAOA;EACE,UAAU;AACZ"`;

exports[`css: sourcesContent 1`] = `
Array [
  "<template>
  <p class=\\"css\\">&lt;css&gt;</p>
  <p :class=\\"$style['css-module']\\">&lt;css&gt; module</p>
  <p class=\\"css-scoped\\">&lt;css&gt; scoped</p>
</template>

<style>
.css {
  color: red;
}
</style>

<style module>
.css-module {
  color: red;
}
</style>

<style scoped>
.css-scoped {
  color: red;
}
</style>
",
]
`;

exports[`less with additionalData: mappings 1`] = `"AAKA;EACE"`;

exports[`less with additionalData: sourcesContent 1`] = `
Array [
  "<template>
  <p class=\\"less\\">&lt;less&gt; with additionalData</p>
</template>

<style lang=\\"less\\">
.less {
  color: @color;
}
</style>
",
]
`;

exports[`sass with import: mappings 1`] = `"AAAA;EACE;;ACOF;EACE"`;

exports[`sass with import: sourcesContent 1`] = `
Array [
  ".sass-with-import-imported
  color: red
",
  "<template>
  <p class=\\"sass-with-import\\">&lt;sass&gt; with import</p>
  <p class=\\"sass-with-import-imported\\">&lt;sass&gt; with import (imported)</p>
</template>

<style lang=\\"sass\\">
@import './sassWithImportImported'

.sass-with-import
  color: red
</style>
",
]
`;

exports[`sass: mappings 1`] = `"AAKA;EACE"`;

exports[`sass: sourcesContent 1`] = `
Array [
  "<template>
  <p class=\\"sass\\">&lt;sass&gt;</p>
</template>

<style lang=\\"sass\\">
.sass
  color: red
</style>
",
]
`;

exports[`src imported sass: mappings 1`] = `"AAAA;EACE;;ACCF;EACE"`;

exports[`src imported sass: sourcesContent 1`] = `
Array [
  ".src-import-sass-imported
  color: red
",
  "@import './src-import-imported'

.src-import-sass
  color: red
",
]
`;

exports[`src imported: mappings 1`] = `"AAAA;EACE,UAAU;AACZ"`;

exports[`src imported: sourcesContent 1`] = `
Array [
  ".src-import {
  color: red;
}
",
]
`;
